x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/build_asset_name_filter.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { buildPhraseFilter, type Filter } from '@kbn/es-query';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { HOST_FIELD } from '../../../../../common/constants';
+import type { InfraCustomDashboardAssetType } from '../../../../../common/custom_dashboards';
+
+const fieldByAssetType = {
+  host: HOST_FIELD,
+} as Record<InfraCustomDashboardAssetType, string>;
+
+export function getFilterByAssetName(
+  assetName: string,
```

Comment:
As we start working with other assets, we'll have to use `assetId`. For hosts `assetId` and `assetName` are `host.name`, that's why everything now works, but that's not the case for other assets.

Reply:
Right, thanks for pointing that out, I changed it to `assetId` 👍